doc/developer-guide/testing.rst

```diff
@@ -0,0 +1,89 @@
+.. _dev-testing:
+
+Testing sites
+=============
+
+It is possible to create end-to-end integration tests for Zotonic
+websites.  Tests like these are called *sitetests*. They run within
+the Zotonic shell, starting the website under test using a special
+database schema so that the main database is not affected by the
+tests.
+
+To run the sitetests for a site called ``example``, run the following command::
+
+  z_sitetest:run(example).
+
+This will stop the ``example`` site, create a new database schema
+(called ``z_sitetest``), start the site (which installs all site data
+into the new schema), and then scans all compiled Erlang modules for
```

Comment:
Perhaps link to [module versioning](http://zotonic.com/docs/latest/developer-guide/modules.html#module-versioning) here.